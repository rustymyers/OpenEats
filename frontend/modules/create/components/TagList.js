import React from 'react'
import {measurements} from '../../common/config'

export default React.createClass({
  getInitialState: function() {
    return {
      value: this.props.value || '',
      comp_value: this.arrayify(this.props.value)
    };
  },

  arrayify(value) {
    var dict = [];
    if (value) {
      var tags = value.split(',');
      for (let title in tags) {
        dict.push({'title': tags[title]})
      }
    }
    return dict
  },

  handleChange(event) {
    this.setState({
      value: event.target.value,
      comp_value: this.arrayify(event.target.value)
    });
    if(this.props.change) {
      this.props.change(event.target.name, this.arrayify(event.target.value));
    }
  },

  render: function () {
    return (
      <div className={this.props.size} key={this.props.id}>
        <div className="form-group">
          {this.props.label ? <label>{this.props.label}</label> : null}
          <input type={this.props.type}
                 name={this.props.name}
                 className="form-control"
                 placeholder={this.props.placeholder}
                 value={this.state.value}
                 onChange={this.handleChange}/>
        </div>
      </div>
    )
  }
});
