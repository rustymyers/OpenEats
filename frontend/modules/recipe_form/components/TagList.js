import React from 'react'

export default React.createClass({
  getInitialState: function() {
    return {
      tags: this.props.tags || [],
      input: this.unarrayify(this.props.tags || []),
      errors: this.props.errors || false,
    };
  },

  unarrayify(value) {
    return value.map((tag, key) => {
      return tag.title
    }).join(', ');
  },

  arrayify(value) {
    var dict = [];
    if (value) {
      var tags = value.split(',');
      for (let title in tags) {
        dict.push({'title': tags[title].trim()})
      }
    }
    return dict
  },

  handleChange(event) {
    this.setState({
      tags: this.arrayify(event.target.value),
      input: event.target.value
    });

    if(this.props.change) {
      this.props.change(event.target.name, this.arrayify(event.target.value));
    }
  },

  componentWillReceiveProps: function(nextProps) {
    if (this.props.tags === undefined && this.props.tags != nextProps.tags) {
      this.setState({
        tags: nextProps.tags,
        input: this.unarrayify(nextProps.tags)
      });
    }

    if ('errors' in nextProps) {
      this.setState({
        errors: nextProps.errors
      });
    }
  },

  render: function () {
    let className = "form-group";
    let errorMessage = false;
    if (this.state.errors !== false) {
      className += " has-error";
      errorMessage = (
        <span className="help-inline">{ this.state.errors[0] }</span>
      )
    }

    return (
      <div className={this.props.size} key={this.props.id}>
        <div className={ className }>
          {this.props.label ? <label>{this.props.label}</label> : null}
          <input type={this.props.type}
                 name={this.props.name}
                 className="form-control"
                 placeholder={this.props.placeholder}
                 value={this.state.input}
                 onChange={this.handleChange}
                 />
          { errorMessage }
        </div>
      </div>
    )
  }
});
