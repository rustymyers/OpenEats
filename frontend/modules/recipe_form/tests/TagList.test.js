import React from 'react';
import TagList from '../components/TagList';
import renderer from 'react-test-renderer';

test('TagList tests', () => {
  const component = renderer.create(
    <TagList tags={ [{'title': 'testme'}, {'title': 'testmetoo'}] }/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('TagList tests', () => {
  const component = renderer.create(
    <TagList tags={ [{'title': 'There was a problem'}] } errors="O no!"/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
