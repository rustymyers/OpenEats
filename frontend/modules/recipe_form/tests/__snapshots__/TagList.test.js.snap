// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagList tests 1`] = `
<div
  className={undefined}
>
  <div
    className="form-group"
  >
    <input
      className="form-control"
      name={undefined}
      onChange={[Function]}
      placeholder={undefined}
      type={undefined}
      value="testme, testmetoo"
    />
  </div>
</div>
`;

exports[`TagList tests 2`] = `
<div
  className={undefined}
>
  <div
    className="form-group has-error"
  >
    <input
      className="form-control"
      name={undefined}
      onChange={[Function]}
      placeholder={undefined}
      type={undefined}
      value="There was a problem"
    />
    <span
      className="help-inline"
    >
      O
    </span>
  </div>
</div>
`;
